#333 bugfix url params with comma need escape

mukundnathan opened the pull request:
The http spec treats ',' as a reserved subdelim character in an url.
api2go states the following for ',' in query params, "If there are multiple values, you have to separate them with a comma. api2go automatically slices the values for you.".  
Hence if comma has to be treated as integral part of the parameter, escape it with \ or %5c for proper urlencoding.

Commit:
bugfix dl-333 fix comma separated url params with urlencoded escape

lightworks backend bug 333.  if comma is in url params lookups fail.  fix by using urlencoded escape \.

Commit:
#333 url parameters with comma need %5c a \ as  escape

If a query parameter used in a filter has commas it needs to be escaped with a \ or %5c to properly urlencode it.

Commit:
removed a debug file

hybernaut commented on api_test.go:
```diff
@@ -1196,6 +1262,27 @@ var _ = Describe("RestHandler", func() {
+		})
+
+		It("Extracts query parameter with urlencoded escaped comma as one string", func() {
+			urlencoded := strings.Replace(title1, ",", "%5c,", -1)
```
shouldn't this be %5C instead of %5c?

hybernaut reviewed (approved):
good work

mukundnathan commented:
%5c or %5C is treated the same in urlencoding

On Thu, Jan 25, 2018 at 3:02 PM, Brian Del Vecchio <notifications@github.com
> wrote:

> *@hybernaut* commented on this pull request.
> ------------------------------
>
> In api_test.go
> <https://github.com/digitallumens/api2go/pull/2#discussion_r163952839>:
>
> > @@ -1196,6 +1262,27 @@ var _ = Describe("RestHandler", func() {
>  			Expect(api2goReq.QueryParams).To(Equal(map[string][]string{"sort": {"title", "date"}}))
>  		})
>
> +		It("Extracts comma separated query parameters as a slice of strings ", func() {
> +			req, err := http.NewRequest("GET", "/v0/posts?title="+title1, nil)
> +			Expect(err).To(BeNil())
> +			c := &APIContext{}
> +
> +			split := strings.Split(title1, ",")
> +			Expect(len(split)).To(Equal(2))
> +
> +			api2goReq := buildRequest(c, req)
> +			Expect(api2goReq.QueryParams).To(Equal(map[string][]string{"title": {split[0], split[1]}}))
> +		})
> +
> +		It("Extracts query parameter with urlencoded escaped comma as one string", func() {
> +			urlencoded := strings.Replace(title1, ",", "%5c,", -1)
>
> shouldn't this be %5C instead of %5c?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/digitallumens/api2go/pull/2#pullrequestreview-91653232>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AhmLlpoCts0830_eWsA1xnyYaKeoC7Yqks5tON3TgaJpZM4RtQlS>
> .
>
